src/backend/src/main/kotlin/org/icpclive/service/ICPCScoreboardService.kt

```diff
@@ -102,45 +67,24 @@ abstract class ICPCScoreboardService(optimismLevel: OptimismLevel) {
                 }
             }
 
-            return@map when(resultType) {
-                ContestResultType.ICPC -> {
-                    ICPCProblemResult(
-                        runsBeforeFirstOk.withIndex().count { isAddingPenalty(it.value, it.index, problemRuns.size) },
-                        runsBeforeFirstOk.withIndex().count { isPending(it.value, it.index, problemRuns.size) },
-                        okRun != null,
-                        okRun?.isFirstSolvedRun == true,
-                        (okRun ?: runsBeforeFirstOk.lastOrNull())?.time
-                    ).also {
-                        if (it.isSolved) {
-                            solved++
-                            penaltyCalculator.addSolvedProblem(okRun!!.time, it.wrongAttempts)
-                            lastAccepted = max(lastAccepted, okRun.time.inWholeMilliseconds)
-                        }
-                    }
-                }
-                ContestResultType.IOI -> {
-                    val maxScore = if(problemRuns.isNotEmpty()) problemRuns.maxBy { it.score }.score else 0.0f
-
-                    IOIProblemResult(
-                        runsBeforeFirstOk.withIndex().count { isAddingPenalty(it.value, it.index, problemRuns.size) },
-                        runsBeforeFirstOk.withIndex().count { isPending(it.value, it.index, problemRuns.size) },
-                        maxScore,
-                        okRun?.isFirstSolvedRun == true,
-                        (okRun ?: runsBeforeFirstOk.lastOrNull())?.time
-                    ).also {
-                        if (it.score > 0) {
-                            solved += it.score
-                            penaltyCalculator.addSolvedProblem(okRun!!.time, it.wrongAttempts)
-                            lastAccepted = max(lastAccepted, okRun.time.inWholeMilliseconds)
-                        }
-                    }
+            ICPCProblemResult(
+                runsBeforeFirstOk.withIndex().count { isAddingPenalty(it.value, it.index, problemRuns.size) },
+                runsBeforeFirstOk.withIndex().count { isPending(it.value, it.index, problemRuns.size) },
+                okRun != null,
+                (okRun?.result as? ICPCRunResult)?.isFirstToSolveRun == true,
```

Comment:
Will it work correctly if there is AC ran after FTS?

Reply:
Yes, okRun is the first ok run.